articles/question-answering/index.md

```diff
@@ -0,0 +1,89 @@
+
+### Document Processing Module
+The document processing module accepts the reformulated question as its input. The document processing module uses an internal information retrieval (IR) system to map the closest documents to the question presented. The documents are sorted according to their similarity and relevance to the question. The document processing module performs three main tasks.
+
+1. *Retrieve* the set of documents from the IR system.
+2. *Filter* the set of documents obtained from the previous step, and reduce the amount of text in each document to produce a concise answer.
+3. *Order* the documents by similarity and relevance to the question.
+
+### Answer Processing Module
+The final module gets the list of concise documents filtered and ordered by the previous module. The answer processing module considers the set of documents and performs three major tasks:
+
+1. *Identify* statements/answers within the concise set of documents.
+2. *Extract* the relevant answer by choosing the appropriate phrases and words that answer the question. This used to be performed heuristically. *Heuristic algorithms* are a way of solving problems where solving in the best possible way is given importance. Parameters such as cost, complexity in terms of space and time, accuracy, or speed are neglected, and instead, obtaining a solution close to the actual answer is emphasized.
+
+   Heuristic algorithms are usually used to solve [NP-complete problems](https://www.britannica.com/science/NP-complete-problem), where there is no known efficient way to find a solution. However, it can be verified if the answer is known by plugging the parameters into the algorithm. Therefore, developing heuristic algorithms involves going back and forth between the solution produced and the algorithm itself.
+
+   The algorithm is said to perform well when the algorithm's solution is as efficient as possible, given the constraints.
+
+3. *Validation* of the answers obtained by the previous step. This is usually used during the design of such question-answering systems to evaluate the outputs obtained. During deployment, the evaluation metrics are reduced to achieve high data throughput from the system.
+
+
+### Evaluating the answers obtained
+The criteria can vary from paper to paper. But more generally, we want the question answering system to produce relevant, correct, and complete answers to the point. Hence many evaluation metrics were developed to measure such ambiguous terminologies. A few of those metrics are F1-scores, precision, recall, etc. For a detailed insight into evaluation metrics, refer to this [article](/engineering-education/evaluating-ml-model-performance/).
+
+Transformers at scale for question answering & neural search. Using NLP via a modular Retriever-Reader-Pipeline. Supporting DPR, Elasticsearch, HuggingFace's Modelhub
```

Comment:
@lalith1403 this section seems incomplete. Could you please clarify/refine these references?

Reply:
My bad. That was a personal note I had pasted here for reference. Will delete it.